### PR TITLE
show compose box before report is ready after logout

### PR DESCRIPTION
**Pull request by getusha**
@techievivek @sobitneupane
### Fixed Issues

$ https://github.com/Expensify/App/issues/16908
PROPOSAL: N/A

### Tests

1. Logout from the app
2. Login with a working account
3. While the skeleton UI is loaded for the chat screen, verify that the compose box is present.
4. Ensure that you can't enter anything in the composer
5. Verify that the Add actions, Emoji Icon and Send icon are disabled.

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Login with a working account
2. Disconnect the device from internet
3. While the skeleton UI is loaded for the whole screen, verify that the compose box is present.
4. Ensure that you can't enter anything in the composer
5. Verify that the Add actions, Emoji Icon and Send icon are disabled.

### QA Steps

Same as Tests
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/75031127/229817325-f05dd5bd-c9d6-4564-a7ed-22f67a55487f.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/75031127/229818164-fc7dda07-0510-4222-a576-655b72e0ba43.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/75031127/229819210-4ef19adc-88c1-49cd-b689-6017aec15d00.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/75031127/229834375-e88c311b-2dae-4d06-b8ac-6c9b3d9b0c5a.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/75031127/229860211-ac1234e5-0fec-48fd-9f1e-299eb5310c03.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/75031127/229847690-c3ce8947-95a9-45ac-a165-7b7780f159df.mov

</details>

### CODE REVIEW

**Comment by mananjadhav** [comment]
@johnmlee101 Does this need C+ review?

**Comment by johnmlee101** [comment]
Yes please!

**Comment by mananjadhav** [comment]
@johnmlee101 @getusha while the composer loads, I think we should also be able to type the chats right?

> The compose box is accessible and a message can be typed even while the skeleton UI loads

Expected result from https://github.com/Expensify/App/issues/14223

https://user-images.githubusercontent.com/3069065/229920462-759065a8-f8e2-4ae5-9504-9586e2b94cea.mov

**Comment by mananjadhav** [comment]
I definitely agree it shouldn't be enabled. I am just highlighting it was the expected result in the linked issue, and we should update the tests accordingly and ensure it is not reported as a regression later.

File: src/pages/home/ReportScreen.js [hunk 1]
```diff
@@ -321,7 +321,10 @@ class ReportScreen extends React.Component {
                             {/* Note: The report should be allowed to mount even if the initial report actions are not loaded. If we prevent rendering the report while they are loading then
                             we'll unnecessarily unmount the ReportActionsView which will clear the new marker lines initial state. */}
                             {(!this.isReportReadyForDisplay() || isLoadingInitialReportActions || isLoading) && (
-                                <ReportActionsSkeletonView containerHeight={this.state.skeletonViewContainerHeight} />
+                                <>
+                                    <ReportActionsSkeletonView containerHeight={this.state.skeletonViewContainerHeight} />
+                                    <EmojiPicker ref={EmojiPickerAction.emojiPickerRef} />
```
Comment (mananjadhav) [inline src/pages/home/ReportScreen.js:326]: I didn't understand why we should move this in this block? Also I see the linked issue is closed.

Comment (mananjadhav) [reply src/pages/home/ReportScreen.js:326]: As I mentioned in this [comment](https://github.com/Expensify/App/pull/16922#issuecomment-1496630341), I think the right solution is not to move this inside an if block, rather remove two differences.

This still isn't convincing why it wouldn't open in offline mode. Hence, I think it is better to remove linking the issue, and if required tackle it separately if we're able to reproduce it consistently.

**Comment by mananjadhav** [comment]
@getusha Let's improve the content for our Tests:

> 1. Logout from the app
> 2. Login with a working account
> 3. While the skeleton UI is loaded for the chat screen, verify that the compose box is present.
> 4. Ensure that you can't enter anything in the composer
> 5. Verify that the Add actions, Emoji Icon and Send icon are disabled.

I think we should also update our Offline tests? QA could be same as Tests

**Comment by mananjadhav** [comment]
Yes I think it makes sense to add an offline test. The one you posted [here](https://github.com/Expensify/App/pull/16922#issuecomment-1496645633) looks fine

@getusha also just confirming are we not removing the two instances of EmojiPicker?

**Comment by mananjadhav** [comment]
Yes thats what I feel, because we introduced the second instance of EmojiPicker, in the same [PR](https://github.com/Expensify/App/pull/16705), for which we're solving the regression.

**Comment by mananjadhav** [comment]
Apologies for the confusion. Let me clarify, I am requesting we fix the multiple EmojiPicker issues in the component tree in this PR because the linked PR introduced it.

Whether that solves the EmojiPicker in Offline mode is a separate issue, which can be first reproduced reliably, and/or kept on Hold (currently it is closed).

So yes we should just ensure we fix the changes introduced by the regression PR. Makes sense?

**Review by mananjadhav — APPROVED** [review]
Thanks for the changes and the patience here. Code looks good, will test now.

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [ ] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/3069065/229938264-f800c86e-f455-493c-abb4-504264e9e272.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/3069065/229938411-58024d89-feb4-445c-a95f-869e708da8b9.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/3069065/229938436-fa682490-7345-4b1f-a9d9-1430f40eb7ab.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/3069065/229938498-a9cdafed-f715-4590-844b-2c76b1b4109f.mov

</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>

Uploading android-composer-when-chat-loading.mov…

</details>


-----

@francoisl I have some issues with the iOS setup. I've completed the checklist for all items except iOS Native. Could you help here?

**Review by thienlnam — COMMENTED** [review]
Taking over review from @johnmlee101 since he's offline right now and it's a deploy blocker

**Comment by mananjadhav** [comment]
@getusha Thanks for the link. It is definitely related to XCode 14.3 but the workaround didn't help. My issue was due to `React-codegen` dependency not compatible with 14.3. I've managed to fix it and updated the screencast.

 @thienlnam All yours.

**Comment by thienlnam** [comment]
Oh perfect, I was working on fixing my simulator to test but this works as well!

**Review by thienlnam — APPROVED** [review]

**Comment by sobitneupane** [comment]
Thanks @mananjadhav @getusha for handling the blocker.

**Pull request merged**